Update UG for Edit Transaction and Summary Commands (#136)
AY2324S1-CS2103-T16-3/tp: docs/UserGuide.md

> > <img src="images/unicash/command-outputs/addTransactionFailedOutput3.png" width="1000" />

#### 4.1.2 EditTransactionCommand
Edits an existing transaction in UniCa$h.

Command: `edit_transaction INDEX [n/NAME] [type/TYPE] [amt/AMOUNT] [dt/DATETIME] [l/LOCATION] [c/CATEGORY]`

Command Options:

> | Option Name | Optional? | Purpose                                                                                       |

Reviewer: Maybe you can add a remark column here on the right, and specify the constraints

Author: Yeah, I think this is a good idea. I've implemented this fix and moved most of the constraints to this additional column to reduce the number of "Important notes"!

|-------------|-----------|-----------------------------------------------------------------------------------------------|
| INDEX       | No        | (One-based) index of the transaction to edit.                                                 |
| n/          | Yes       | Name of the transaction.                                                                      |
| type/       | Yes       | Transaction type of transaction.<br/>Valid types are `income` and `expense`.                  |
| amt/        | Yes       | Monetary amount of transaction. Has to be a positive value.                                   |
| dt/         | Yes       | Date and time where transaction was made.<br/>Defaults to current date time if not specified. |
| l/          | Yes       | Location where transaction was made.<br/>Defaults to `''` if not specified.                   |
| c/          | Yes       | Category tagged to that transaction.<br/>No categories tagged if not specified.               |

> Important notes:

Reviewer: Maybe this would be better named as constraints? While it is good to be detailed, this section is a bit too long and could be quite overwhelming for users. I would suggest either moving most of these to an appendix, or have group sections for clarity.

Author: I've followed your suggestion in the other comment to move these constraints to a new "Remark" column!

> 1. There is a character limit for `Name` and `Location` set at up to 500 characters.

Reviewer: ```suggestion
1. `Name` and `Location` have a 500 character limit.
```

2. `Amount` entered has to be positive for both `income` and `expense`.
3. `Amount` is automatically rounded to 2 decimal places.

> 4. `Datetime` entered can be in the `dd-MM-uuuu HH:mm` (e.g. 31-10-2023 12:00), `uuuu-MM-dd HH:mm` (e.g. 2023-10-31 22:59), or `dd MMM uuuu HH:mm` (e.g. 31 Oct 2023 22:59) formats.

Reviewer: Maybe we can add this in an appendix, because this datetime format applies to more than just the `editTransaction` command.

Author: Agreed! 😄  I was thinking of creating a section to list out the constraints for each field, since after the discussion today, it seems that there have been more constraints implemented. However, should we leave it as it is for the PE dry run tomorrow, and work on it in the coming week for v1.4?

5. `Category` entered will be converted and stored as all lowercase.
6. Each transaction is able to have a maximum of 5 categories.
7. There is a character limit for `Category` set at up to 15 characters.
8. While all options besides `INDEX` are optional, **you must specify at least one field** you wish to edit (i.e. `Name`, `Type`, `Amount`, `Datetime`, `Location`, or `Category`).

> 9. You must provide values for `Name`, `Type`, and `Amount`; they cannot be left empty.

Reviewer: Is this required, if you already listed this in the table and stated it as optional/not optional?

Author: Regarding the constraints for points 6 and 7, I have followed your suggestion to move them to the new "Remark" column.

For point 8, the reason I included this point was to emphasize that the user should not expect the command to work if they don't specify any field despite the fact that all fields have been labelled as "Optional". They will need to specify at least one field.

For point 9, the point I was trying to drive across was that the user should not attempt to make `Name`, `Type`, and `Amount` empty. They can make datetime, location, or category empty as it will be reset to default, but this is not possible for the `Name`, `Type`, and `Amount` fields. I have slightly modified this line to make the point clearer. Let me know what you think! 😄 
In particular, I changed it to:
"You must provide a non-empty value for the `Name`, `Type`, and `Amount` fields if you wish to edit them; they cannot be left empty."

#### 4.3.2 SummaryCommand

Displays a summary of the expenses saved in UniCa$h.

Command: `summary`

Important notes:

> 1. If you have no expenses saved in UniCa$h, the summary pop-up window will not appear. However, if you have previously opened the summary window and then remove all expenses, it will remain open until you manually close it.

Reviewer: Maybe you can label this as _Expected Behaviour_
```suggestion
Expected Behaviour: The summary of expenses is a pop-up window. If there are no expenses saved in UniCa$h, the pop-up window will not appear. Do note that the summary window will remain open should you decide to remove all expenses after it is opened, until you manually close it.
```
